detection_rules/mixins.py

```diff
@@ -0,0 +1,51 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+"""Generic mixin classes."""
+from typing import TypeVar, Type
+
+import marshmallow_dataclass
+
+from .utils import cached
+
+T = TypeVar('T')
+ClassT = TypeVar('ClassT')  # bound=dataclass?
+
+
+def _strip_none_from_dict(obj: T) -> T:
+    """Strip none values from a dict recursively."""
+    if isinstance(obj, dict):
+        return {key: _strip_none_from_dict(value) for key, value in obj.items() if value is not None}
+    if isinstance(obj, list):
+        return [_strip_none_from_dict(o) for o in obj]
+    if isinstance(obj, tuple):
+        return tuple(_strip_none_from_dict(list(obj)))
+    return obj
+
+
+class MarshmallowDataclassMixin:
```

Comment:
so I don't want to do that, because dataclass already constructs a metaclass and multiple metaclasses in the hierarchy gets funky.

that would also turn this into an inheritance model when what I think we want is a Mixin model. I just want to add a few methods but leave everything else untouched